--- ../third_party/sunfounder/server/motor.py	2018-05-30 03:55:23.000000000 +0200
+++ sunfounder-patched/motor.py	2018-06-17 21:07:29.240727831 +0200
@@ -27,7 +27,7 @@
 # ===========================================================================
 def setSpeed(speed):
 	speed *= 40
-	print 'speed is: ', speed
+	#print 'speed is: ', speed
 	pwm.write(EN_M0, 0, speed)
 	pwm.write(EN_M1, 0, speed)
 
@@ -76,7 +76,7 @@
 		GPIO.output(Motor0_A, GPIO.HIGH)
 		GPIO.output(Motor0_B, GPIO.LOW)
 	else:
-		print 'Config Error'
+		print('Config Error')
 
 def motor1(x):
 	if x == 'True':
@@ -120,11 +120,11 @@
 		elif direction == -1:  # Backward
 			backward()
 		else:
-			print 'Argument error! direction must be 1 or -1.'
+			print('Argument error! direction must be 1 or -1.')
 	elif status == 0: # Stop
 		stop()
 	else:
-		print 'Argument error! status must be 0 or 1.'
+		print('Argument error! status must be 0 or 1.')
 
 def test():
 	while True:
